rapid/src/main/java/com/vrg/rapid/MembershipView.java

```diff
@@ -176,32 +212,52 @@ void ringDelete(final Endpoint node) {
         Objects.requireNonNull(node);
         rwLock.readLock().lock();
         try {
-            if (!rings.get(0).contains(node)) {
+            if (!allNodes.contains(node)) {
                 throw new NodeNotInRingException(node);
             }
-
-            if (rings.get(0).size() <= 1) {
-                return Collections.emptyList();
+            if (!cachedObservers.containsKey(node)) {
+                cachedObservers.put(node, computeObserversOf(node));
```

Comment:
This writes with a read lock acquired, so it might not actually guard the memory